ast/src/input/program_state.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the Leo library.
+
+// The Leo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Leo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+
+/// Processed Program state.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct ProgramState {
+    pub state: Definitions,
+}
+
+impl TryFrom<ParsedInputFile> for ProgramState {
```

Comment:
@collinc97 is this necessary? I thought we were deprecating state files?

Reply:
It is likely but we haven't confirmed yet